lib/c/evercrypt_targetconfig.h

```diff
@@ -18,11 +18,13 @@
 #define TARGET_ARCHITECTURE_ID_SYSTEMZ 5
 #define TARGET_ARCHITECTURE_ID_POWERPC64 6
 
+#if defined(__has_include)
 #if __has_include("config.h")
 #include "config.h"
 #else
 #define TARGET_ARCHITECTURE TARGET_ARCHITECTURE_ID_UNKNOWN
```

Comment:
So `TARGET_ARCHITECTURE` isn't defined at all when `__has_include` is not available? We might want this defined in any case where we don't have the `config.h`.

Reply:
I'm wondering if we should just get rid of TARGET_ARCHITECTURE. It doesn't seem to provide much added value and brings us further away from our goal of thinking exclusively in terms of features (rather than in terms of specific targets)

Reply:
I wouldn't mind getting rid of it. But this PR changes when and how it gets added. I think we should keep the logic here and then think about removing it.

Reply:
agreed that this should be a separate PR... it's just that we don't use that thing anywhere else (as far as I can tell) except for hacl_vec128_not_avx () and we could refactor that code to not depend on it...